test/integration/lighter_than_air_dynamics.cc

```diff
@@ -0,0 +1,118 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <gtest/gtest.h>
+#include <string.h>
+#include <vector>
+
+#include <gz/msgs/double.pb.h>
+
+#include <gz/common/Console.hh>
+#include <gz/common/Util.hh>
+#include <gz/transport/Node.hh>
+#include <gz/utils/ExtraTestMacros.hh>
+
+#include "gz/sim/Link.hh"
+#include "gz/sim/Model.hh"
+#include "gz/sim/Server.hh"
+#include "gz/sim/SystemLoader.hh"
+#include "gz/sim/TestFixture.hh"
+#include "gz/sim/Util.hh"
+#include "gz/sim/World.hh"
+
+#include "test_config.hh"
+#include "../helpers/EnvTestFixture.hh"
+
+using namespace gz;
+using namespace sim;
+
+class LighterThanAirDynamicsTest : public InternalFixture<::testing::Test>
+{
+  /// \brief Test a world file
+  /// \param[in] _world Path to world file
+  /// \param[in] _namespace Namespace for topic
+  public: std::vector<math::Vector3d> TestWorld(const std::string &_world,
+   const std::string &_namespace);
+
+  public: math::Vector3d defaultForce{0, 0, 10.0};
+};
+
+//////////////////////////////////////////////////
+std::vector<math::Vector3d> LighterThanAirDynamicsTest::TestWorld(
+  const std::string &_world, const std::string &_namespace)
+{
+  // Maximum verbosity for debugging
+  common::Console::SetVerbosity(4);
+
+  // Start server
+  ServerConfig serverConfig;
+  serverConfig.SetSdfFile(_world);
+
+  TestFixture fixture(serverConfig);
+
+  Model model;
+  Link body;
+  std::vector<math::Vector3d> bodyVels;
+  fixture.
+  OnConfigure(
+    [&](const Entity &_worldEntity,
+      const std::shared_ptr<const sdf::Element> &/*_sdf*/,
+      EntityComponentManager &_ecm,
+      EventManager &/*eventMgr*/)
+    {
+      World world(_worldEntity);
+
+      auto modelEntity =  world.ModelByName(_ecm, _namespace);
+      EXPECT_NE(modelEntity, kNullEntity);
+      model = Model(modelEntity);
+
+      auto bodyEntity = model.LinkByName(_ecm, _namespace + "_link");
+      EXPECT_NE(bodyEntity, kNullEntity);
+
+      body = Link(bodyEntity);
+      body.EnableVelocityChecks(_ecm);
+
+    }).
+  OnPostUpdate([&](const UpdateInfo &/*_info*/,
+                            const EntityComponentManager &_ecm)
+    {
+      auto bodyVel = body.WorldLinearVelocity(_ecm);
+      ASSERT_TRUE(bodyVel);
+      bodyVels.push_back(bodyVel.value());
+    }).
+  Finalize();
+
+  fixture.Server()->Run(true, 2000, false);
+  EXPECT_EQ(2000u, bodyVels.size());
+
+  EXPECT_NE(model.Entity(), kNullEntity);
+  EXPECT_NE(body.Entity(), kNullEntity);
```

Comment:
Sorry for the incredibly long feedback time. But this test does not seem to be checking anything about the behaviour of the dynamics. It'll be hard to spot regressions if we dont test that.

Reply:
I agree that the tests are limited which I have implemented. 
Will have a look at this more closely! 
Thanks for the feedback.

Reply:
What we do for hydrodynamics is push a model and then try to measure its terminal velocity. Perhaps this is an approach which you want to adopt?

Reply:
Added tests to check the instability of pitch and yaw.